zephyr/CMakeLists.txt

```diff
@@ -1129,9 +1129,15 @@ zephyr_library_sources_ifdef(CONFIG_WAVES_CODEC_STUB
 	${SOF_AUDIO_PATH}/module_adapter/module/waves/maxx_stub.c
 )
 
-zephyr_library_sources_ifdef(CONFIG_PROBE
-	${SOF_SRC_PATH}/probe/probe.c
-)
+if(CONFIG_PROBE STREQUAL "m")
+	add_subdirectory(${SOF_SRC_PATH}/probe/llext
+			 ${PROJECT_BINARY_DIR}/probe_llext)
+	add_dependencies(app probe)
+elseif(CONFIG_PROBE)
+	zephyr_library_sources_ifdef(CONFIG_PROBE
```

Comment:
nitpick: no need for the `_ifdef` version here any more (check other module entries): just use `zephyr_library_sources()`